### ISSUE TITLE
UnsatisfiedDependencyException Ambiguous argument values for parameter of type [com.arangodb.springframework.core.ArangoOperations]

### ISSUE THREAD

**Issue by hberndt** [arangodb/spring-data#14]
Using spring-boot 1.5.10.RELEASE and arangodb-spring-data 1.0.1 we get the following error 
```
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'contractRepository': Unsatisfied dependency expressed through constructor parameter 1: Ambiguous argument values for parameter of type [com.arangodb.springframework.core.ArangoOperations] - did you specify the correct bean references as arguments?
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:736)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:189)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1193)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1095)
	at ...
```

**Comment by Sukrat** [comment]
I am getting the same error. 
`compile group: 'org.springframework.boot', name: 'spring-boot-devtools', version: '1.5.10.RELEASE'`
This dependency in the project seems to trigger it. Do you have this dependency as well. I want to know the work around to it as well.

**Comment by mpv1989** [comment]
Thank you both for reporting. I'll take a look.

**Comment by mark--** [comment]
We have the same problem. See https://stackoverflow.com/questions/49303080/spring-boot-application-context-with-arangodb-repository-cannot-be-created-if-us for a detailed description and a minimal example project. See the discussion for https://github.com/spring-projects/spring-boot/issues/6541 for a possible explanation.

**Comment by mpv1989** [comment]
@hberndt @Sukrat @mark-- 
I pushed a fix which worked for me when using `spring-boot-devtools`. Would be nice if you can test it in your environment too. The fix is in `1.0.2-SNAPSHOT` and `2.0.3-SNAPSHOT`. Please let me know your results, then I can release a new version.

Keep in mind, that you have to add the staging repository of oss.sonatype.org in your project when using one of our snapshots.
```XML
<repositories>
  <repository>
    <id>arangodb-snapshots</id>
    <url>https://oss.sonatype.org/content/groups/staging</url>
  </repository>
</repositories>
```

Thanks & best
Mark

**Comment by mark--** [comment]
We tested version 1.0.2-SNAPSHOT and Spring Boot 1.5.6.Release and it works!
We tested with and without devtools and `@MockBean` with and without name Parameter.
Thank you for the quick response!

**Comment by Sukrat** [comment]
@mpv1989 I have tested the 1.0.2-SNAPSHOT with Spring Boot 1.5.10.Release and it works fine!

**Comment by mpv1989** [comment]
released versions `1.0.2` and `2.0.3`.
